### PR TITLE
MakeScaffoldGeneric isotope removal

### LINKED ISSUE #6836: MurckoScaffold.MakeScaffoldGeneric() has issues with isotopes
**Describe the bug**
When using MakeScaffoldGeneric(), if the input molecule has explicit isotopes, this is retained in the generic scaffold. Even when the atom is not carbon. This is a less obscure problem than it seems, because e.g. ChEMBL has dozens of molecules in there with explicit isotopes (which is how I discovered this) within the CSK.

**To Reproduce**
```python
from rdkit import Chem
from rdkit.Chem.Scaffolds import MurckoScaffold

smi1="[14CH2]1CCC1"
smi2="[15NH]1CCC1"
smi3="[235U]1CCC1"
mols=[Chem.MolFromSmiles(smi) for smi in [smi1,smi2,smi3]]

for m in mols:
    csk=MurckoScaffold.MakeScaffoldGeneric(MurckoScaffold.GetScaffoldForMol(m))
    print(Chem.MolToSmiles(csk))
```

this outputs:

```
C1C[14CH2]C1
C1C[15CH2]C1
C1C[235CH2]C1
```

**Expected behavior**
No isotope flags. Expected output:
```
C1CCC1
C1CCC1
C1CCC1
```

**Configuration (please complete the following information):**
 - RDKit version:  2023.03.3
 - OS: Ubuntu 20.04
 - Python version (if relevant): not relevant
 - Are you using conda? yes
 - If you are using conda, which channel did you install the rdkit from? conda-forge
 - If you are not using conda: how did you install the RDKit? n/a

**Additional context**
The fix is probably to add `atom.SetIsotope(0)` after line 40 in the murcko script. https://github.com/rdkit/rdkit/blob/master/rdkit/Chem/Scaffolds/MurckoScaffold.py#L40
Just posting the issue here to make sure this makes sense before doing a pull request.

### PR DESCRIPTION
**Pull request by dehaenw**
#### Reference Issue
Fixes #6836 

#### What does this implement/fix? Explain your changes.
reseting the isotope to 0 for every atom to fix unintended retention of isotope labels during generation of generic scaffolds

### CODE REVIEW

**Review by greglandrum — CHANGES_REQUESTED** [review]
@dehaenw thanks for this!
The changes you have made so far look good, but we also always include tests with each bug fix to ensure that the fix actually works. In this case it would be adding a new test at the bottom of ScaffoldNetwork/catch_tests.cpp (use the existing tests as templates) as well as one at the bottom of rdkit/Chem/Scaffolds/UnitTestMurckoScaffold.py

It would also be great if you could extend this PR to fix the similar problem in MolHash/hashfunctions.cpp:ExtendedMurckoScaffold and add a test for that as well in MolHash/catch_tests.cpp

Please let me know have questions about any of these.

**Comment by dehaenw** [comment]
apologies for the several commits. was working in a clumsy way fixing the tests. tests should now work, I hope they are formatted ok, I also added an isotope reset in the AnonymousGraph part of MolHash/hashfunctions.cpp

File: rdkit/Chem/Draw/github4763.svg
Comment (greglandrum) [inline rdkit/Chem/Draw/github4763.svg:1]: I don't think this file is part of this PR

**Review by greglandrum — CHANGES_REQUESTED** [review]
There's one extra file in the PR that shouldn't be there.
Once that's removed, this is fine.

File: rdkit/Chem/Draw/github4763.svg
Comment (dehaenw) [reply rdkit/Chem/Draw/github4763.svg:1]: Ah, whoops! Deleting

**Review by greglandrum — APPROVED** [review]
LGTM
Thanks for the contribution @dehaenw!

**Pull request merged**